### PR TITLE
Necessary low level elements for User Preferences

### PR DESCRIPTION
**Pull request by sgrossberndt**
* New OptimisationMethods
* Walking and Cycle Profile search
* IncludeAlternativeOptions
For Trips

Documentation chapter 5.4.  https://vdvde.sharepoint.com/sites/CENTC278WG3SG8OJP/Freigegebene%20Dokumente/General/Version_2.0/210713-OJP-64_e_stm-2.0_work.docx?web=1

### CODE REVIEW

**Comment by sgrossberndt** [comment]
superseeds: https://github.com/VDVde/OJP/pull/221/files
done incorrectly in https://github.com/VDVde/OJP/pull/271, new try here

**Review by ue71603 — DISMISSED** [review]

**Review by skinkie — APPROVED** [review]

File: OJP/OJP_Trips.xsd [hunk 1]
```diff
@@ -106,11 +106,14 @@
...
-			<xs:element name="OptimisationMethod" type="OptimisationMethodEnumeration" minOccurs="0">
-				<xs:annotation>
-					<xs:documentation>the types of algorithm that can be used for planning a journey (fastest, least walking, etc)</xs:documentation>
-				</xs:annotation>
-			</xs:element>
+			<xs:choice>
+				<xs:element name="OptimisationMethod" type="OptimisationMethodEnumeration"/>
+				<xs:element name="OptimisationMethods" type="OptimisationMethodList" minOccurs="0">
+					<xs:annotation>
+						<xs:documentation>the types of algorithm that can be used for planning a journey (fastest, least walking, etc)</xs:documentation>
```
Comment (skinkie) [inline OJP/OJP_Trips.xsd:113]:
```suggestion
						<xs:documentation>the types of algorithm that can be used for planning a journey (fastest, least walking, etc.)</xs:documentation>
```

File: OJP/OJP_Trips.xsd [hunk 2]
```diff
@@ -172,13 +180,69 @@
...
+		<xs:annotation>
+			<xs:documentation>List of OptimisationMethods. The order of precedence - if used at all - is high to low. In single criteria optimisers the optimisation methods may be used in parallel and not truely used to score the results.</xs:documentation>
+		</xs:annotation>
+		<xs:sequence>
+			<xs:element name="OptimisationMethod" type="OptimisationMethodEnumeration" maxOccurs="unbounded"/>
+		</xs:sequence>
+	</xs:complexType>
 	<xs:simpleType name="OptimisationMethodEnumeration">
 		<xs:annotation>
 			<xs:documentation>the types of algorithm that can be used for planning a journey (fastest, least walking, etc). If multiple optimisation methods are needed, then several requests should be performed in parallel.</xs:documentation>
```
Comment (skinkie) [inline OJP/OJP_Trips.xsd:248]:
```suggestion
			<xs:documentation>the types of algorithm that can be used for planning a journey (fastest, least walking, etc.). If multiple optimisation methods are needed, then several requests should be performed in parallel.</xs:documentation>
```

File: OJP/OJP_Trips.xsd [hunk 3]
```diff
@@ -106,11 +106,14 @@
...
-			<xs:element name="OptimisationMethod" type="OptimisationMethodEnumeration" minOccurs="0">
-				<xs:annotation>
-					<xs:documentation>the types of algorithm that can be used for planning a journey (fastest, least walking, etc)</xs:documentation>
-				</xs:annotation>
-			</xs:element>
+			<xs:choice>
+				<xs:element name="OptimisationMethod" type="OptimisationMethodEnumeration"/>
+				<xs:element name="OptimisationMethods" type="OptimisationMethodList" minOccurs="0">
+					<xs:annotation>
+						<xs:documentation>The types of algorithm that can be used for planning a journey (fastest, least walking, etc)</xs:documentation>
```
Comment (skinkie) [inline OJP/OJP_Trips.xsd:113]:
```suggestion
						<xs:documentation>The types of algorithm that can be used for planning a journey (fastest, least walking, etc.)</xs:documentation>
```

**Review by skinkie — DISMISSED** [review]

**Review by ue71603 — APPROVED** [review]
@skinkie  ready for your approval.

**Review by skinkie — APPROVED** [review]

**Pull request merged**